manifests/warranty.pp

```diff
@@ -5,12 +5,17 @@
 # Used by the fact in this module
 #
 class dell::warranty (
-  $api_key,
+  Optional[String] $api_key   = undef,
```

Comment:
I'm not sure of the need to make the `$api_key` optional, when you could just not include the class

Reply:
I'm guessing you want to always include it and then pass the key through hiera?

Reply:
I made it optional so it wouldn't break backward compatibility.  If someone were to include the class previously and not provide a key, it would still compile.  If just set to `String $api_key = undef`, or `String $api_key`, compilation would now fail.